### PR TITLE
Add `--caching` option (off by default)

### LINKED ISSUE #103: Make caching optional
ATM local filesystem caching of accessed content is non-optional. To make fsspec work multi-threaded, heavy locking was introduced and we are using custom branch with that.  That locking renders access very slow. As in many use cases network access could be almost as fast as the drive access or even faster (e.g. on an EC2 instance), it might make sense to make such caching optional, or even not enabled by default at all so we could use non-patched fsspec with default installation of datalad-fuse.

So let's add `--caching none,ondisk` with `none` as default, and provide documentation in e.g. README to say that patched version of fsspec is needed for `ondisk`.

NB while coming up with interface, might be worth considering that later we might make caching support configurable as well, so "ondisk" might not be descriptive enough. But we could think about it whenever we get there.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #103.

### CODE REVIEW

**Comment by yarikoptic** [comment]
Great, thank you!  could you please 

- look into why benchmarking installation fails?
- also add parametrize to e.g. `test_fuse` on caching so we generally test both implementations?

File: datalad_fuse/tests/test_fuse.py [hunk 1]
```diff
@@ -74,7 +77,9 @@ def test_fuse_subdataset(tmp_path, superdataset, cache_clear, transparent, tmp_h
         dots = [".datalad", ".git", ".gitattributes"]
     else:
         dots = [".datalad", ".gitattributes"]
-    with fusing(ds.path, tmp_path, transparent=transparent, wait=True) as mount:
+    with fusing(
+        ds.path, tmp_path, caching=True, transparent=transparent, wait=True
+    ) as mount:
```
Comment (yarikoptic) [inline datalad_fuse/tests/test_fuse.py:82]: do you have ideas why codecov claims that we are loosing now coverage  of the `cache-clear`?
https://app.codecov.io/gh/datalad/datalad-fuse/pull/104/blob/datalad_fuse/fsspec_cache_clear.py

Comment (jwodder) [reply datalad_fuse/tests/test_fuse.py:82]: I don't know.  It doesn't appear we ever tested that command to begin with.

Comment (yarikoptic) [reply datalad_fuse/tests/test_fuse.py:82]: <details>
<summary>indeed... odd... I reran with this patch and did not spot traceback from being printed :-/</summary> 

```patch
diff --git a/datalad_fuse/fsspec_cache_clear.py b/datalad_fuse/fsspec_cache_clear.py
index 60fcfbb..10d726b 100644
--- a/datalad_fuse/fsspec_cache_clear.py
+++ b/datalad_fuse/fsspec_cache_clear.py
@@ -1,3 +1,5 @@
+import sys
+import traceback
 from typing import Any, Dict, Iterator, Optional
 
 from datalad.distribution.dataset import (
@@ -42,6 +44,7 @@ class FsspecCacheClear(Interface):
     def __call__(
         dataset: Optional[Dataset] = None, recursive: bool = False
     ) -> Iterator[Dict[str, Any]]:
+        traceback.print_stack(file=sys.stderr)
         ds = require_dataset(
             dataset, purpose="clear fsspec cache", check_installed=True
         )

```
</details>

**Pull request merged**